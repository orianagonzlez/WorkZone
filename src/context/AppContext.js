import React from "react";
import { useCallback } from "react";
import { fetchToken } from "../helpers/getData";

const { createContext, useState, useEffect } = require("react");

export const AppContext = createContext();
//   {
//   user: {
//     email: "",
//     id: "",
//     nombre: "",
//     isLogged: false,
//     token: "",
//     checking: true,
//   },
// }

const initialState = {
  email: "",
  id: "",
  nombre: "",
  isLogged: false,
  token: "",
  checking: true,
};

export const AppProvider = ({ children }) => {
  const [user, setUser] = useState(initialState);

  const verifyToken = useCallback(async () => {
    const token = localStorage.getItem("token");
    // Si token no existe
    if (!token) {
      setUser({
        email: "",
        id: "",
        nombre: "",
        isLogged: false,
        token: "",
        checking: false,
      });

      return false;
    }

    const resp = await fetchToken(
      "https://workzone-backend-mdb.herokuapp.com/api/auth/renew"
    );
    if (resp.ok) {
      localStorage.setItem("token", resp.token);
      const { user } = resp;
      const { email, uid, nombre, apellido, fechaNacimiento, username } =
        resp.data;

      setUser({
        email: email,
        id: uid,
        nombre: `${nombre} ${apellido}`,
        username: username,
        fechaNacimiento: fechaNacimiento,
        isLogged: true,
        checking: false,
      });

      return true;
    } else {
      setUser({
        email: "",
        id: "",
        nombre: "",
        isLogged: false,
        token: "",
        checking: false,
      });

      return false;
    }
  }, []);

  return (
    <AppContext.Provider
      value={{
        user,
        setUser,
        verifyToken,
      }}
    >
      {children}
    </AppContext.Provider>
  );
};
