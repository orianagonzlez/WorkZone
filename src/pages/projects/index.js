import React from 'react';
import Projects from '../../components/projects/Projects';
import Layout from '../../components/layout/Layout';

export default function ProjectsPage() {
    return (
        <Layout>
            <Projects />
        </Layout>
    )
}