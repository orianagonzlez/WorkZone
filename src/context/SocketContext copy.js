import React, { useContext, useEffect } from "react";
import { createContext } from "react";
import { useSocket } from "../hooks/useSocket";
import { AppContext } from "./AppContext";
import { ChatContext } from "./ChatContext";

export const SocketContext = createContext();

export const SocketProvider = ({ children }) => {
  const { socket, online, connectSocket, disconnectSocket } = useSocket(
    //"https://workzone-backend-mdb.herokuapp.com/"
    "http://localhost:8080/"
  );

  const { chat, setChat } = useContext(ChatContext);

  const { user } = useContext(AppContext);

  //Escuchar cuando se conecta
  useEffect(() => {
    if (user.isLogged) {
      connectSocket();
    }
  }, [user, connectSocket]);

  //Escuchar cuando se desconecta
  useEffect(() => {
    if (!user.isLogged) {
      disconnectSocket();
    }
  }, [user, disconnectSocket]);

  //Escuchar los cambios en los usuarios conectados

  useEffect(() => {
    socket?.on("lista-usuarios", (usuarios) => {
      setChat({ ...chat, projects: usuarios });
    });
  }, [socket, setChat, chat]);

  useEffect(() => {
    socket?.on("mensaje", (msg) => {

      // if (chat.activeChat === msg.para) {
      //   console.log(chat, "aqui debo tener al menos 1 ");
      //   setChat({
      //     ...chat,
      //     messages: [...chat.messages, msg],
      //   });
      // } else {
      //   //MANDAR NOTIFICACION
      //   console.log("esta entrando mal el msj --------------");
      // }
    });
  }, [socket, chat]);

  return (
    <SocketContext.Provider value={{ socket, online }}>
      {children}
    </SocketContext.Provider>
  );
};
