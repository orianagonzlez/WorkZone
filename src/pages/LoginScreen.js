import React, { useContext } from 'react'
import { Container, Form, Button, Col } from 'react-bootstrap'
import { Link } from 'react-router-dom';
import { AppContext } from '../context/AppContext';
import { postData } from '../helpers/postData';
import { useForm } from '../hooks/useForm';

export const LoginScreen = () => {
  
    const {setUser, user} = useContext(AppContext);

    const [formValues, handleInputChange] = useForm({
        email: '',
        password: ''
    });

    const { email, password } = formValues;

    const handleLogin = (e) => {
        e.preventDefault();
        console.log("Login Sirve")
        console.log(email, password)

        let body = {
          email: email, 
          contrasena: password
          }
        
        postData('users/login', body).then( r => {
            console.log('me respondio' + r);
            if (r.status === 'success') {
              const {email, id_usuario, nombre, apellido, fecha_nacimiento, username } = r.data;
              setUser({
                  ...user,
                  email: email,
                  id: id_usuario,
                  nombre: `${nombre} ${apellido}`,
                  username: username,
                  fechaNacimiento: fecha_nacimiento,
                  isLogged: true,
                });
            } else {
              console.log('error');
            }
        });
        
    }


    return (
        <div className="login_main">

            <Container className="login_container">

                <h1 className="auth_title">¡BIENVENIDO DE VUELTA!</h1>
                
                <Form className="login_form" onSubmit={handleLogin}>

                    <Form.Row className="d-flex align-items-center justify-content-start">

                        <Form.Group as={Col}>
                            <Form.Label>Correo electrónico</Form.Label>
                            <Form.Control className="input"
                                type="text"
                                placeholder="example@gmail.com"
                                name="email"
                                autoComplete="off"
                                value={email}
                                onChange={handleInputChange} />
                        </Form.Group>

                    </Form.Row>
                    <Form.Row className="d-flex align-items-center justify-content-start">

                        <Form.Group as={Col}>
                            <Form.Label>Contraseña</Form.Label>
                            <Form.Control className="input"
                                type="password"
                                placeholder="**********"
                                name="password"
                                value={password}
                                onChange={handleInputChange} />
                        </Form.Group>

                    </Form.Row>


                        <Container className="justify-content-center">

                            <Form.Group>
                                <div className="auth_link">
                                    <Link
                                        to="/auth/register"
                                        className="link" >
                                        ¿Todavía no tienes cuenta? ¡Registrate!
                                </Link>
                                </div>

                            </Form.Group>


                            <div className="button">
                                <Button
                                    className="auth_button"
                                    variant="primary"
                                    type="submit"
                                >
                                    INGRESAR
                                </Button>
            
                            </div>


                            <Form.Group>
                                <div className="auth_link"> ¿Olvidaste tu contraseña?</div>
                            </Form.Group>

                        </Container>

                </Form>
                
            </Container>

        </div>
    )
}
