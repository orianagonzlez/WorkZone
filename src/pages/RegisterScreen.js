import React, { useContext, useState } from "react";
import { Container, Form, Button, Col } from "react-bootstrap";
import { Link } from "react-router-dom";
import Swal from "sweetalert2";
import { AppContext } from "../context/AppContext";
import { postData } from "../helpers/postData";
import { useForm } from "../hooks/useForm";

export const RegisterScreen = () => {
  const [disabled, setDisabled] = useState(false);
  const { setUser, user } = useContext(AppContext);

  const [formValues, handleInputChange] = useForm({
    name: "",
    lastname: "",
    email: "",
    username: "",
    password: "",
    password2: "",
    birthday: "",
  });

  const {
    name,
    lastname,
    email,
    username,
    password,
    password2,
    birthday,
  } = formValues;

  const handleRegister = (e) => {
    setDisabled(true);
    e.preventDefault();


    if (isFormValid()) {

      let body = {
        nombre: name[0].toUpperCase() + name.slice(1),
        apellido: lastname[0].toUpperCase() + lastname.slice(1),
        contrasena: password,
        fechaNacimiento: new Date(birthday),
        email: email,
        username: username,
      };
      const url = "https://workzone-backend-mdb.herokuapp.com/api/auth/create";
      postData(url, body).then((r) => {
        if (r.ok) {
          localStorage.setItem("token", r.token);
          const {
            email,
            uid,
            nombre,
            apellido,
            fechaNacimiento,
            username,
          } = r.data;
          setUser({
            ...user,
            email: email,
            id: uid,
            nombre: `${nombre} ${apellido}`,
            username: username,
            fechaNacimiento: fechaNacimiento,
            isLogged: true,
          });
        } else {

          Swal.fire({
            icon: "error",
            title: "Oops...",
            text: "Credenciales Invalidas",
            confirmButtonColor: "#22B4DE",
          });
        }
      });
    }
    setDisabled(false);
  };

  const isFormValid = () => {
    //TODO Falta verificar todos los campos

    if (name.trim().length === 0 || lastname.trim().length === 0
       || email.trim().length === 0 || name.trim().length === 0
       || birthday.trim().length === 0 || username.trim().length === 0) {
      Swal.fire({
        icon: "error",
        title: "Oops...",
        text: "Por favor ingrese todos los campos",
        confirmButtonColor: "#22B4DE",
      });
      return false;
    } else if (password !== password2 || password.trim().length < 5) {
      Swal.fire({
        icon: "error",
        title: "Oops...",
        text: "Por favor ingrese una clave mayor a 5 digitos y que coincida con la confirmación de clave.",
        confirmButtonColor: "#22B4DE",
      });
      return false;
    }
    return true;
  };

  return (
    <div className="register_main ">
      <Container className="register_container  justify-content-center">
        <h1 className="auth_title text-center">¡BIENVENIDO A WORK ZONE!</h1>
        <h2 className="auth_subtitle">Información Personal</h2>

        <Form className="register_form" onSubmit={handleRegister}>
          <Form.Row className="d-flex  align-items-center justify-content-around">
            <Form.Group as={Col} md={5} sm={12} className="input_container">
              <Form.Label className="input_label">Nombre</Form.Label>
              <Form.Control
                className="input"
                type="text"
                name="name"
                autoComplete="off"
                value={name}
                onChange={handleInputChange}
              />
            </Form.Group>

            <Form.Group as={Col} md={5} sm={12} className="input_container">
              <Form.Label className="input_label">Apellido</Form.Label>
              <Form.Control
                className="input"
                type="text"
                name="lastname"
                autoComplete="off"
                value={lastname}
                onChange={handleInputChange}
              />
            </Form.Group>
          </Form.Row>

          <Form.Row className="d-flex  align-items-center justify-content-around">
            <Form.Group as={Col} md={5} sm={12} className="input_container">
              <Form.Label className="input_label">
                Fecha de Nacimiento
              </Form.Label>
              <Form.Control
                className="input"
                type="date"
                name="birthday"
                value={birthday}
                onChange={handleInputChange}
              />
            </Form.Group>
            <Form.Group as={Col} md={5} sm={12} className="input_container">
              <Form.Label className="input_label">
                Correo Electrónico
              </Form.Label>
              <Form.Control
                className="input"
                type="text"
                name="email"
                autoComplete="off"
                value={email}
                onChange={handleInputChange}
              />
            </Form.Group>
          </Form.Row>

          <Form.Row className="d-flex  align-items-center justify-content-around">
            <Form.Group
              as={Col}
              md={5}
              sm={12}
              className="input_container justify-content-around "
            >
              <Form.Label className="input_label">Contraseña</Form.Label>
              <Form.Control
                className="input"
                type="password"
                name="password"
                value={password}
                onChange={handleInputChange}
              />
            </Form.Group>
            <Form.Group as={Col} md={5} sm={12} className="input_container">
              <Form.Label className="input_label">
                Confirme su Contraseña
              </Form.Label>
              <Form.Control
                className="input"
                type="password"
                name="password2"
                value={password2}
                onChange={handleInputChange}
              />
            </Form.Group>
          </Form.Row>
          <Form.Row className="d-flex align-items-center justify-content-around">
            <Form.Group as={Col} md={5} sm={12} className="input_container ">
              <Form.Label className="input_label">Usuario</Form.Label>
              <Form.Control
                className="input"
                type="text"
                name="username"
                autoComplete="off"
                value={username}
                onChange={handleInputChange}
              />
            </Form.Group>
          </Form.Row>

          <Form.Group>
            <div className="auth_link">
              <Link to="/auth/login" className="link">
                ¿Ya tienes cuenta? ¡Inicia Sesión!
              </Link>
            </div>
          </Form.Group>
          <div className="button">
            <Button className="auth_button" variant="primary" type="submit" disabled={disabled}>
              REGISTRAR
            </Button>
          </div>
        </Form>
      </Container>
    </div>
  );
};
