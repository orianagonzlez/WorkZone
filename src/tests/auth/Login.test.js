import { mount, shallow } from "enzyme";
import React from "react";
import { LoginScreen } from "../../pages/LoginScreen";

import { AppProvider } from "../../context/AppContext";
import { SocketProvider } from "../../context/SocketContext";
import { ChatProvider } from "../../context/ChatContext";
import { TimerProvider } from "../../context/TimerContext";
import { MemoryRouter } from "react-router";

describe("Probando login", () => {
  test("Debe mostrarse correctamente", () => {
    const appContextValue = {
      setUser: jest.fn(),
      user: {
        email: "prueba@gmail.com",
        id: "12345678",
        nombre: "prueba myTest",
        isLogged: true,
        token: "drftyvubndkds31321wepjfjduei",
        checking: false,
      },
    };

    const chatContextValue = {
      chat: {
        activeChat: "",
        projects: [],
        messages: [],
        notification: "",
        project: "",
      },
      dispatch: jest.fn(),
    };

    const wrapper = mount(
      <MemoryRouter>
        <AppProvider value={appContextValue}>
          <TimerProvider>
            <LoginScreen />
          </TimerProvider>
        </AppProvider>
      </MemoryRouter>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
