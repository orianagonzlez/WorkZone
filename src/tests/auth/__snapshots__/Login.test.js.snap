// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando login Debe mostrarse correctamente 1`] = `
<div
  className="login_main"
>
  <div
    className="login_container container"
  >
    <h2
      className="auth_title"
    >
      ¡WorkZone!
    </h2>
    <h3
      className="auth_subtitle"
    >
      Trabajar a distancia nunca había sido tan fácil
    </h3>
    <form
      className="login_form"
      onSubmit={[Function]}
    >
      <div
        className="d-flex align-items-center justify-content-start form-row"
      >
        <div
          className="form-group col"
        >
          <label
            className="form-label"
          >
            Correo electrónico
          </label>
          <input
            autoComplete="off"
            className="input form-control"
            name="email"
            onChange={[Function]}
            placeholder="example@gmail.com"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        className="d-flex align-items-center justify-content-start form-row"
      >
        <div
          className="form-group col"
        >
          <label
            className="form-label"
          >
            Contraseña
          </label>
          <input
            className="input form-control"
            name="password"
            onChange={[Function]}
            placeholder="**********"
            type="password"
            value=""
          />
        </div>
      </div>
      <div
        className="justify-content-center container"
      >
        <div
          className="form-group"
        >
          <div
            className="auth_link"
          >
            <a
              className="link"
              href="/register"
              onClick={[Function]}
            >
              ¿Todavía no tienes cuenta? ¡Registrate!
            </a>
          </div>
        </div>
        <div
          className="button"
        >
          <button
            className="auth_button btn btn-primary"
            disabled={false}
            type="submit"
          >
            INGRESAR
          </button>
        </div>
        <div
          className="form-group"
        >
          <div
            className="auth_link"
            onClick={[Function]}
          >
             
            ¿Olvidaste tu contraseña?
          </div>
        </div>
      </div>
    </form>
  </div>
</div>
`;
